sympy/matrices/common.py

```diff
@@ -816,37 +834,76 @@ def jordan_block(kls, *args, **kwargs):
         >>> Matrix.jordan_block(4, x, band='lower')
         Matrix([
         [x, 0, 0, 0],
         [1, x, 0, 0],
         [0, 1, x, 0],
         [0, 0, 1, x]])
+
+        Creating a Jordan block with keyword arguments
+
         >>> Matrix.jordan_block(size=4, eigenvalue=x)
         Matrix([
         [x, 1, 0, 0],
         [0, x, 1, 0],
         [0, 0, x, 1],
         [0, 0, 0, x]])
-        """
 
-        klass = kwargs.get('cls', kls)
-        size, eigenvalue = None, None
-        if len(args) == 2:
-            size, eigenvalue = args
-        elif len(args) == 1:
-            size = args[0]
-        elif len(args) != 0:
-            raise ValueError("'jordan_block' accepts 0, 1, or 2 arguments, not {}".format(len(args)))
-        rows, cols = kwargs.get('rows', None), kwargs.get('cols', None)
-        size = kwargs.get('size', size)
-        band = kwargs.get('band', 'upper')
-        # allow for a shortened form of `eigenvalue`
-        eigenvalue = kwargs.get('eigenval', eigenvalue)
-        eigenvalue = kwargs.get('eigenvalue', eigenvalue)
+        Creating a rectangular Jordan block:
+
+        >>> Matrix.jordan_block(rows=4, cols=3, eigenvalue=x)
+        Matrix([
+        [x, 1, 0],
+        [0, x, 1],
+        [0, 0, x],
+        [0, 0, 0]])
+
+        Notes
+        =====
+        The keyword arguments ``size``, ``rows``, ``cols`` relates to
+        the Jordan block size specifications.
+
+        If you want to create a square Jordan block, specify either
+        one of the three arguments.
+
+        If you want to create a rectangular Jordan block, specify
+        ``rows`` and ``cols`` individually.
+
+        +--------------------------------+---------------------+
+        |        Arguments Given         |     Matrix Shape    |
+        +----------+----------+----------+----------+----------+
+        |   size   |   rows   |   cols   |   rows   |   cols   |
+        +==========+==========+==========+==========+==========+
+        |   size   |         Any         |   size   |   size   |
+        +----------+----------+----------+----------+----------+
+        |          |        None         |     ValueError      |
+        |          +----------+----------+----------+----------+
+        |   None   |   rows   |   None   |   rows   |   rows   |
+        |          +----------+----------+----------+----------+
+        |          |   None   |   cols   |   cols   |   cols   |
+        +          +----------+----------+----------+----------+
+        |          |   rows   |   cols   |   rows   |   cols   |
+        +----------+----------+----------+----------+----------+
+
+        References
+        ==========
+
+        https://en.wikipedia.org/wiki/Jordan_matrix
+        """
+        klass = kls if cls is None else cls
 
         if eigenvalue is None:
-            raise ValueError("Must supply an eigenvalue")
+            # allow for a shortened form of `eigenvalue`
+            eigenval = kwargs.get('eigenval', None)
```

Comment:
Are there other methods where `eigenval` is used as keyword argument? (I think it is OK, just curious.)

Reply:
I don't think any keywords such as `eigenval` or `eigenvalue` are already used by other function.

However, I think there is not much point of supporting abbreviations or such, in library functions, and if possible, only one keyword should be mainly used and other should be dropped.
Either if it is eigenval or eigenvalue

Reply:
-1: preferrably one way (from `import this`)

If a user wants to shorten things up they can define

EIG = 'eigenvalue' and call it with `(...EIG=x0)`; let the user be creative and Sympy predictable (in that, as a rule, we don't try to guess which keyword you are giving).

Reply:
@smichr 
I also had agreed on a point that the library code should not be some naive natural language processor,
But from #15256, I think some people have other point of view about deprecating some old keywords.

ping @moorepants

Reply:
Oh...I see that it allowed this before. So let's leave it non-breaking and just note that this should be avoided in the future.

Reply:
However, I think some people may prefer `eigenval` than `eigenvalue`, or we may have to check other keywords to check whether sympy prefers using some mild abbreviation, than full math notation.